python/src/iceberg/utils/schema_conversion.py

```diff
@@ -0,0 +1,274 @@
+            ...     ),
+            ...     NestedField(
+            ...         field_id=507,
+            ...         name="partitions",
+            ...         field_type=StructType(
+            ...             NestedField(
+            ...                 field_id=509,
+            ...                 name="contains_null",
+            ...                 field_type=BooleanType(),
+            ...                 is_optional=False,
+            ...                 doc="True if any file has a null partition value",
+            ...             ),
+            ...             NestedField(
+            ...                 field_id=518,
+            ...                 name="contains_nan",
+            ...                 field_type=BooleanType(),
+            ...                 is_optional=True,
+            ...                 doc="True if any file has a nan partition value",
+            ...             ),
+            ...         ),
+            ...         is_optional=True,
+            ...         doc="Summary for each partition",
+            ...     ),
+            ...     schema_id=1
+            ... )
+            >>> avro_schema == iceberg_schema
+            True
+
+        Args:
+            field:
+
+        Returns:
+
+        """
+        field_id = field["field-id"]
+        field_name = field["name"]
+        field_doc = field.get("doc")
+
+        avro_type, is_optional = self._resolve_union(field["type"])
+        if isinstance(avro_type, dict):
+            avro_type = avro_type["type"]
+
+        if isinstance(avro_type, str) and avro_type in AvroSchemaConversion.PRIMITIVE_FIELD_TYPE_MAP:
+            return NestedField(
+                field_id=field_id,
+                name=field_name,
+                field_type=AvroSchemaConversion.PRIMITIVE_FIELD_TYPE_MAP[avro_type],
+                is_optional=is_optional,
+                doc=field_doc,
+            )
+        elif avro_type == "record":
+            return NestedField(
+                field_id=field_id, name=field_name, field_type=self._parse_record(field), is_optional=is_optional, doc=field_doc
+            )
+        elif avro_type == "array":
+            inner_type, _ = self._resolve_union(field["type"])
+            assert isinstance(inner_type, dict)
+            inner_type["items"]["field-id"] = inner_type["element-id"]
+            inner_field = self._parse_field(inner_type["items"])
+
+            return NestedField(
+                field_id=field_id, name=field_name, field_type=inner_field.type, is_optional=is_optional, doc=field_doc
+            )
+        elif avro_type == "map":
+            return NestedField(
+                field_id=field_id,
+                name=field_name,
+                field_type=MapType(
+                    key_id=field["key-id"],
+                    key_type=StringType(),
+                    value_id=field["value-id"],
+                    value_type=self._parse_field(field["values"]),
```

Comment:
I think there are two problems here. First, this returns a field so you probably want to get the `type` from it. Second, `field["values"]` could be a string, which would cause conversion to a field to fail.

Reply:
Fixed both of them 👍🏻